File: 0901-online-stock-span/0901-online-stock-span.cpp
class StockSpanner {
private:
    vector<int> prices;
    stack<int> s;
    int day;
public:
    StockSpanner() {
        day = 0;
    }
    int next(int price) {
        prices.push_back(price);
        while(!s.empty() && prices[s.top()] <= price){
            s.pop();
        }
        int span;
        if(s.empty()){
            span = day+1;
        }else{
            span = day-s.top();
        }
        s.push(day);
        day++;
        return span;
    }
};

File: 0901-online-stock-span/0901-online-stock-span.py
class StockSpanner:

    def __init__(self):
        self.prices = []
        self.s = []
        self.day = 0

    def next(self, price: int) -> int:
        self.prices.append(price)
        while self.s and self.prices[self.s[-1]] <= price:
            self.s.pop()
        if not self.s:
            span = self.day+1
        else:
            span = self.day-self.s[-1]
        self.s.append(self.day)
        self.day += 1
        return span


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)